index.js

```diff
@@ -1,18 +1,90 @@
+start()
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log('Welcome to my Guessing Game!\n' + '*Games alternate after intital selection*\n')
+  let gameChoice = await ask('To Start Please Choose Either Computer Guesses or Human Guesses\n')
+  gameChoice = gameChoice.trim().toLowerCase()
+  if (gameChoice == 'computer guesses') {
+    computerGuesses()
+  }
+  if (gameChoice == 'human guesses') {
+    humanGuesses()
+  }
 }
+// Computer Guess Game Function
+async function computerGuesses() {
+  console.log('\nYou have now entered Mode: Computer Guesses\n' + 'To play this game please choose a numerical range for the game(ex. 1 to 100)')
+  let userMinRange = parseInt(await ask('What is your desired low end range?\n'));
+  let userMaxRange = parseInt(await ask('What is your desired top end range?\n'));
+  console.log('You entered: ' + userMinRange + ' to ' + userMaxRange)
+  let computerGenNum = randomNum(userMinRange, userMaxRange)
+  console.log("Please choose a secret number within your chosen range now and remember it")
+  let answer = await ask('Is your number ' + computerGenNum + ' (yes / no)' + '?\n')
+
+  while (answer !== 'yes') {
+    let highLow = await ask('Is your number lower or higher or correct?\n');
+    highLow = highLow.trim().toLowerCase()
+    if (highLow === 'higher') {
+      userMinRange = computerGenNum + 1
+      computerGenNum = midValue(userMinRange, userMaxRange)
+      console.log('My next guess is ' + computerGenNum + '\n')
+    } else if (highLow === 'lower') {
+      userMaxRange = computerGenNum - 1
+      computerGenNum = midValue(userMinRange, userMaxRange)
+      console.log('My next guess is ' + computerGenNum + '\n')
+    } else if (highLow === 'correct') {
+      console.log('I have Won!\n' + 'Skynet Activated')
+      humanGuesses()
+    } else {
+      console.log('Unrecognized input, Try Again')
+    } if (userMaxRange < userMinRange) {
+      console.log('Cheater! Start Over')
+      process.exit()
+    }
+  } if (answer.toLowerCase() === 'yes') {
+    console.log('I have Won!\n' + 'Skynet Activated')
+    humanGuesses()
+  }
+}
+// Human Guess Game Function
+async function humanGuesses() {
+  console.log('\nYou have now entered Mode: Human Guesses\n' + 'To play this game please choose a numerical range for the game(ex. 1 to 100)')
+  let userMinRange = parseInt(await ask('What is your desired low end range?\n'));
+  let userMaxRange = parseInt(await ask('What is your desired top end range?\n'));
+  console.log('You entered: ' + userMinRange + ' to ' + userMaxRange)
+  let computerGenNum = randomNum(userMinRange, userMaxRange)
+  console.log("I have chosen a secret number within your range for you to guess,")
+
```

Comment:
There is nothing currently stopping the player from guessing outside of the given range.  Consider modifying the logic to reflect that. Admittedly, that's on the player for choosing a range and then ignoring it, but the enforcement of rules is part of creating a bulletproof program!